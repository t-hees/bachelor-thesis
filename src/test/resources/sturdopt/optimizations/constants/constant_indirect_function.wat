(module
  (type $func_type (func (param i32 i32) (result i32)))
  (table 2 funcref)
  (elem (i32.const 0) $const_func $return_3)
  (func $const_func (type $func_type) ;; This type must remain $func_type after optimization since it is in a table
    local.get 0
    local.get 1
    i32.add
  )
  (func $return_3 (result i32)
    i32.const 3
  )
  (func (export "main") (result i32)
    (call_indirect (result i32) (i32.const 1))
    i32.const 9
    (call_indirect (type $func_type) (i32.const 0))
  )
)
