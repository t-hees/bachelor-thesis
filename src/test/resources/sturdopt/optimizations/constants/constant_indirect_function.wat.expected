(module
  (type $func_type (func (param i32 i32) (result i32)))
  (table 2 funcref)
  (elem (i32.const 0) $const_func $return_3)
  (func $const_func (type $func_type) ;; This type must remain $func_type after optimization since it is in a table
    i32.const 3
    i32.const 9
    drop
    drop
    i32.const 12
  )
  (func $return_3 (result i32)
    i32.const 3
  )
  (func (export "main") (result i32)
    i32.const 1 ;; table idx
    drop
    i32.const 3

    i32.const 9

    i32.const 0 ;; table idx
    drop
    drop
    drop
    i32.const 12
  )
)
